python/cuml/test/test_pickle.py

```diff
@@ -350,6 +366,58 @@ def test_unfit_pickle(model_name):
     assert mod_unpickled is not None
 
 
+def modified_clone(estimator, *, safe=True):
+    """
+    Identical to sklearn.clone except it uses equality comparison for numeric
+    types instead of "is"
+    """
+    estimator_type = type(estimator)
+    # XXX: not handling dictionaries
+    if estimator_type in (list, tuple, set, frozenset):
+        return estimator_type([clone(e, safe=safe) for e in estimator])
+    elif not hasattr(estimator, 'get_params') or isinstance(estimator, type):
+        if not safe:
+            return copy.deepcopy(estimator)
+        else:
+            if isinstance(estimator, type):
+                raise TypeError("Cannot clone object. " +
+                                "You should provide an instance of " +
+                                "scikit-learn estimator instead of a class.")
```

Comment:
why scikit-learn? Should this be cuml.Base?

Reply:
Will rewrite this method.

Reply:
Rewritten